### PR TITLE
Create employee

### PR DESCRIPTION
**Pull request by shivamchouhan123**

### CODE REVIEW

File: app/controllers/api/v1/employees_controller.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ module V1
     class EmployeesController < ApplicationController
       before_action :authenticate_employee!
       before_action :set_employee, only: %i[show]
-      skip_before_action :verify_authenticity_token, only: [:update_password]
+      skip_before_action :verify_authenticity_token #, only: [:update_password]
```
Comment (ror-mahesh) [inline app/controllers/api/v1/employees_controller.rb:8]: why you change this?

File: app/controllers/api/v1/employees_controller.rb [hunk 2]
```diff
@@ -18,7 +18,23 @@ def index
...
+       def create
+        if current_employee.is_hr?
+          employee_data = Employee.new(employee_params)
+          if employee_data.save
+            render json: {
+             data: serializer_data(employee_data, employee_serializer),
+             message: ['Successfully'],status: 201, type: 'Success'
+            }
+          else
+             render json: {errors: [employee_data.errors.full_messages]}, status: 302, type: 'Found'
```
Comment (ror-mahesh) [inline app/controllers/api/v1/employees_controller.rb:31]: is this send proper error messages?

File: app/controllers/api/v1/employees_controller.rb [hunk 3]
```diff
@@ -47,8 +63,12 @@ def set_employee
         render json: { message: 'Not Found' }, status: 404 unless @employee_data.present?
       end
 
+      # def employee_params
+      #   params.require(:employee).permit(:email, :password, :password_confirmation, :current_password)
+      # end
+
       def employee_params
-        params.require(:employee).permit(:email, :password, :password_confirmation, :current_password)
+        params.require(:employee).permit(:id, :name, :email, :password, :phone_number, :password_confirmation, :role_id, :designation_id,)
```
Comment (ror-mahesh) [inline app/controllers/api/v1/employees_controller.rb:71]: where is image attribute?

Are you permitted all attributes of employee?

**Review by ror-mahesh — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/daily_tasks_controller.rb [hunk 4]
```diff
@@ -9,13 +9,15 @@ class DailyTasksController < ApplicationController
...
+        if current_employee.is_hr?   
+         daily_task_data= DailyTask.all.order(id: :desc)  
+         employee = daily_task_data.where(employee_id: params[:employee_id]) 
           render json: {
-           data: serializer_data(daily_task_data, daily_task_serializer),
-           message: ['show daily task'], status: 200, type: 'Success'
-          }
-         else
+            data: serializer_data(daily_task_data, daily_task_serializer),
+            data: single_data(employee, daily_task_serializer),
```
Comment (ror-mahesh) [inline app/controllers/api/v1/daily_tasks_controller.rb:17]: Is This correct way to do this type of tasks?

**Review by ror-mahesh — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/daily_tasks_controller.rb (hunk 4 above)
Comment (shivamchouhan123) [reply app/controllers/api/v1/daily_tasks_controller.rb]: resolve

Comment (shivamchouhan123) [reply app/controllers/api/v1/daily_tasks_controller.rb]: check

**Review by ror-mahesh — APPROVED** [review]
LGTM

**Pull request merged**